### PR TITLE
[Perf] avoid cpu op in QwenImageCrossAttention

### PR DESCRIPTION
**Pull request by ZJY0516**
## Purpose
avoid cpu op in `QwenImageCrossAttention`, e.g. `tensor.all()`

And we should put attention metadata preparation out of transformer blocks

image

## Test

```shell
python text_to_image.py --model Qwen/Qwen-Image-2512  --prompt "A futuristic cyberpunk city at night, neon lights reflecting on wet streets, highly detailed, 8k"  --negative_prompt "cartoon"  --width 1024  --height 1024  --num_inference_steps 50  --guidance_scale 4.0  --seed 42
```

55.265s ->  54.4743s

### non-divisble length (padding and attention_mask)
```
python examples/offline_inference/text_to_image/text_to_image.py --height 1024 --width 1536 --ulysses_degree 4
```
13.8639s -> 12.9673s

| height | width |  ulysses_degree | output_image |
| ---| ---| ---| --- |
|  1328 | 1328 | 4 |  qwen_image_output|
|  1024 | 1024 | 1 |  qwen_image_output|
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
@gcanlin @tjtanaa please check whether it helps in NPU and ROCm

**Comment by ZJY0516** [comment]
cc @wtomin

**Comment by ZJY0516** [comment]
~~do not merge this PR, as it appears to be introducing an error when SDPA is used.~~

**Comment by kliuae** [comment]
This is a great optimization for ROCm. On MI300X, it improves Qwen-Image e2e latency from 30.7s to 20.4s.

**Comment by hsliuustc0106** [comment]
@ZJY0516 do we still encounter the acc problem?

**Comment by ZJY0516** [comment]
> @ZJY0516 do we still encounter the acc problem?

It's ready now
